src/components/ImageUploader/src/ImageUploader.vue

```diff
@@ -0,0 +1,311 @@
+			if (!this.canUploadImage) {
+				return NO_MORE_IMAGES_COUNT;
+			}
+
+			return this.maxImages - this.images.length;
+		},
+
+		canUploadMultiple() {
+			// eslint-disable-next-line no-magic-numbers
+			return !this.maxImages || this.maxImages > 1;
+		},
+	},
+
+	methods: {
+		/**
+		 * Processes images selected by the picker, ensuring the number of images
+		 * is not above the max, passes them to the formatter, and then triggers the upload.
+		 *
+		 * @param {Array} images a list of Files
+		 */
+		selectImages(images) {
+			if (!this.canUploadImage) {
+				return;
+			}
+
+			const imagesToUpload = images.length > this.remainingImagesCount
+				// eslint-disable-next-line no-magic-numbers
+				? images.slice(0, this.remainingImagesCount)
+				: images;
+
+			const formattedImages = this.formatImages(imagesToUpload);
+			this.$emit('image-uploader:input', [...this.images, ...formattedImages]);
+			formattedImages.forEach((image) => this.handleImageUpload(image));
+		},
+
+		/**
+		 * Transforms a list of image Files into a standard object with status/upload attributes
+		 *
+		 * @param {Array} images a list of Files
+		 */
+		formatImages(images) {
+			const formattedImages = images
+				.map((image) => {
+					this.nextID += ID_INCREMENT;
+
+					return {
+						id: this.nextID,
+						file: image,
+						status: IMAGE_SELECTOR_STATUSES.PENDING,
+						progress: 0,
+					};
+				});
+			formattedImages.forEach((image) => this.buildImageURL(image));
+
+			return formattedImages;
+		},
+
+		/**
+		 * Starts the upload process for a formatted image. First checks to make sure an image
+		 * fits the filesize limit, and then calls the upload handler function (if present).
+		 *
+		 * @param {Object} image a formatted image object
+		 */
+		async handleImageUpload(image) {
+			this.validateImageSize(image);
+
+			if (!this.isImageValid(image)) {
+				this.setImageFailed(image);
+				return;
+			}
+
+			if (!this.uploadHandlerFn) {
+				this.setImageComplete(image);
+				return;
+			}
+
+			try {
+				const response = await this.uploadHandlerFn({
+					imageFile: image.file,
+					setImageProgress: (progress) => this.setImageProgress(image, progress),
+				});
+				this.setImageComplete(image, response);
+			} catch (error) {
+				this.setImageFailed(image, error);
+			}
+		},
+
+		/**
+		 * Removes an image from the list
+		 *
+		 * @param {String} imageID id of a formatted image object
+		 */
+		removeImage(imageID) {
+			this.$emit('image-uploader:input', this.images.filter(({ id }) => id !== imageID));
+		},
+
+		/**
+		 * Transforms an image File into a base64 URL. Used to display on the UI.
+		 *
+		 * @param {Object} image formatted image object
+		 */
+		async buildImageURL(image) {
+			const url = await new Promise((resolve) => {
+				const reader = new FileReader();
+				reader.onloadend = () => resolve(reader.result);
+				// eslint-disable-next-line unicorn/prefer-add-event-listener
+				reader.onerror = () => resolve('');
```

Comment:
Wouldn't we be missing potential errors here? Why resolve and not reject? https://stackoverflow.com/a/46568146